mongodb/context_strategy.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"time"
+)
+
+// ContextStrategy is an interface that represents strategy for providing contexts for running MongoDB requests
+type ContextStrategy interface {
```

Comment:
This interface is mixing multiple actions of different nature from appending to Counting. It looks like that not implementing it and leaving the implementation to the consumer side might simplify the logic.

Reply:
Inteface instances should provide contexts for corresponding `mongodb.EventStore` method. By default `BackgroundContextStrategy` (do nothing) is used. Another implementation is `TimeoutContextStrategy` cancels queries to mongo (default timeouts are set and can be overriden).

Reply:
This is from CodeRewiew:
```
Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.``` 
https://github.com/golang/go/wiki/CodeReviewComments#interfaces
The interface is simply expressing an action that is exported from a pkg. Adding an interface with such a broader scopes will limit the usability of the package. But it can be used or not so will not hurt in any case. Usually in go an interface is expressing a set of common action with common purpose otherwise a composite interface kicks in to group all the single actions under a common denominator. In any case keeping or removing it will not hurt wanted pin point that this interface looks more fit for an OOP programing language than Golang interfaces :).

Reply:
This is very narrow-purpose interface used only in `mongodb.EventStore`, not the general-puprose one.